### ISSUE TITLE
[Hold] Update DeleteMembersFromWorkspace to work with optimistically creates chats offline

### ISSUE THREAD

**Issue by bondydaa** [Expensify/App#17137]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

Hold for https://github.com/Expensify/App/pull/16764 has been merged to be merged

## Action Performed:
Break down in numbered steps

1. Go offline
2. Add new member to workspace, see optimistic report added to LHN
3. Remove user from workspace, optimistic report stays in LHN
4. Go online
5. Optimistic report in LHN becomes "real chat", if you open that chat it then disappears from LHN and you see it's archived.

## Expected Result:
When you remove a user from a workspace their optimistic expense policy chat should have the "strike through" UI to indicate the pending removal action and once the user goes online the chat should be archived.

Though this bring a question to mind: should the user never actually be added and subsequently archived? I'm am not sure about how this is expected to work with offline first.

## Actual Result:
The chat remains in the LHN and appears as a "normal" chat until the user opens the chat, at that point the chat is loaded and shown as Archived and removed from the LHN.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

N/A

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:** Not yet, will only show after is merged https://github.com/Expensify/App/pull/16764
**Reproducible in production?:** Not yet, will only show after is merged https://github.com/Expensify/App/pull/16764
**If this was caught during regression testing, add the test name, ID and link from TestRail:** n/a
**Email or phone of affected tester (no customers):** n/a
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 n/a
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:** n/a
**Issue reported by:** @bondydaa 
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017ce054ad9e1bebb6</li>
        <li>Upwork Job ID: 1654578579504644096</li>
        <li>Last Price Increase: 2023-05-05</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @garrettmknight (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by bondydaa** [comment]
@garrettmknight this is on hold until https://github.com/Expensify/App/pull/16764 is merged, I just wanted to get it opened so we can add any other bugs related to this API command so we can keep the scope of https://github.com/Expensify/App/pull/16764 focused on just `AddMembersToWorkspace` API command.

**Comment by garrettmknight** [comment]
Sounds good, will check in later this week

**Comment by garrettmknight** [comment]
PR is still open, continuing to hold

**Comment by garrettmknight** [comment]
PR is still open, moving to weekly.

**Comment by garrettmknight** [comment]
PR is still open.

**Comment by garrettmknight** [comment]
@bondydaa Now that the PR we were waiting on has merged, it looks like this is working correctly. Can you try it too and confirm? If it isn't, reassign to me and I'll take it from there.

**Comment by bondydaa** [comment]
oh wow that'd be 💯 if it's just working (maybe someone else fixed it elsewhere or on a different bug GH). 

I'll give it a try and see if it's still there.

**Comment by bondydaa** [comment]
As I was trying to test this last week I discovered the feature it was dependent on was actually broken on staging by a different PR so I tagged that as a deploy blocker and need to follow up with it. Going to throw another `[hold]` on here until we get that all sorted.

**Comment by bondydaa** [comment]
haven't had a chance to dig back in here yet, I will this week and make something actionable.

**Comment by bondydaa** [comment]
Okay had a chance to re-test this I think it it's mostly fixed? 

While offline after I removed the user from the workspace the chat is still in the LHN but they do show as having the strikethrough styling in the member list

<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/dd719d48-3f12-4aad-8cb7-aa05a24d8046">

Then after I sent online the chat was properly removed from the LHN and when I search for it it shows it's archived properly.
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/92c1d5c4-2d84-44c2-bccf-3b1066700508">

Test to see how just removing a user from a workspace while offline (not creating a new one, adding member, removing them all while offline) it looks like it's the same behavior: 

strikethrough for in the workspace member list but LHN policy expense chat stays the same until the user goes back online. 

Garrett is on sabbatical so i'm going to re-add the bug label to get another set of eyes here but I believe we can close this out.

**Comment by bondydaa** [comment]
@greg-schroeder question here is: should we be doing anything optimistically in the LHN for policy expense chats when a user is removed from the workspace while offline? 

My gut says no but just wanted to double check, ignore the OP mostly for now.

**Comment by greg-schroeder** [comment]
@bondydaa honestly the flow in your comment linked [here](https://github.com/Expensify/App/issues/17137#issuecomment-1564964619) seems solid to me - do we need to continue forwards here? I'd be okay closing.

**Comment by bondydaa** [comment]
cool let's close this out then!
